it/controllers/pensions/annualAllowance/PensionProviderTaxPaidAnnualAllowanceControllerISpec.scala

```diff
@@ -0,0 +1,488 @@
+    val amountAboveLimitErrorMessage = "The amount of tax your client’s pension provider paid or agreed to pay must be less than £100,000,000,000"
+    val validAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in the correct format"
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedTitle = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedHeading = "Did your client’s pension schemes pay or agree to pay the tax?"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val amountAboveLimitErrorMessage = "The amount of tax your client’s pension provider paid or agreed to pay must be less than £100,000,000,000"
+    val validAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in the correct format"
+    val noEntryAmountErrorMessage = "Enter the amount paid, or that needs to be paid, in pounds"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  private implicit val url: Int => String = pensionProviderTaxPaidAnnualAllowanceUrl
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      import Selectors._
+      import user.commonExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "render did your pension schemes pay or agree to pay the tax with no pre-filling" which {
+          lazy val result: WSResponse = showPage(user, anPensionsUserDataEmptyCya)
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          radioButtonCheck(yesText, 1, checked = Some(false))
+          radioButtonCheck(noText, 2, checked = Some(false))
+          buttonCheck(buttonText, continueButtonSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          textOnPageCheck(totalNonUkTax, amountText)
+          inputFieldValueCheck(amountInputName, amountInputSelector, "")
+          formPostLinkCheck(pensionProviderTaxPaidAnnualAllowanceUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render page with 'yes' pre-filled and amount field set" which {
```

Comment:
Suggest to rename "... amount field set to a valid amount".

Reply:
Fixed, I renamed the name of the test

Reply:
Confirmed.